core/src/main/python/synapse/ml/core/platform/Platform.py

```diff
@@ -67,3 +67,9 @@ def find_secret(secret_name, keyvault=SECRET_STORE, override=None):
             f"and would like to manually specify your key for Azure KeyVault or Databricks Secrets,"
             f'please add the override="YOUR_KEY_HERE" to the arguments of the find_secret() method'
         )
+
+
+def display_for_synapse_batch_mode(data):
+    if isinstance(data, DataFrame):
+        data.collect()
+    print(data)
```

Comment:
nit: you might want to call the other display in the other part of the if branch to be minimally invasive

Reply:
Can you help me think of a graceful way to do this? I don't want to import the notebookutils.visualization.display in Platform.py as that's a synapse-specific import.

maybe this could work?

```suggestion
def display_for_synapse_batch_mode(data):
    if isinstance(data, DataFrame):
        data.collect()
        print(data)
    elif running_on_synapse() or running_on_synapse_internal()
        from notebookutils.visualization import display
        display(data)
    else
        display(data)
```